File: mobile/lib/services/corporate_service.dart
import 'package:http/http.dart' as http;
import 'dart:convert';

class CorporateTrainingService {
  final String apiBaseUrl;

  CorporateTrainingService(this.apiBaseUrl);

  Future<dynamic> createTraining(Map<String, dynamic> trainingData) async {
    final response = await http.post(
      Uri.parse('$apiBaseUrl/corporate/training'),
      headers: {'Content-Type': 'application/json'},
      body: json.encode(trainingData),
    );
    return json.decode(response.body);
  }
}

File: frontend/src/services/corporateService.js
import api from './api';

export const createTraining = async (trainingData) => {
  try {
    const response = await api.post('/corporate/training', trainingData);
    return response.data;
  } catch (error) {
    console.error('Failed to create training:', error);
    throw new Error(error.response?.data?.message || 'Failed to create training');
  }
};
